cmd/kusk/cmd/install.go

```diff
@@ -106,47 +110,117 @@ var installCmd = &cobra.Command{
 
 		kuskui.PrintStart("installing kusk...")
 
+		c, err := utils.GetK8sClient()
+		if err != nil {
+			reportError(err)
+			return err
+		}
+
 		if err := applyk(dir); err != nil {
 			kuskui.PrintError("❌ failed installing Kusk")
 			reportError(err)
 			return err
 		}
-
 		namespace := "kusk-system"
 		name := "kusk-gateway-manager"
-		c, err := utils.GetK8sClient()
-		if err != nil {
+
+		if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, name, time.Duration(10*time.Minute), "component"); err != nil {
+			kuskui.PrintError("failed installing Kusk")
+			reportError(err)
+			return err
+		}
+
+		if err := utils.WaitForDeploymentReady(cmd.Context(), c, namespace, name, time.Duration(10*time.Minute)); err != nil {
+			kuskui.PrintError("failed installing Kusk")
 			reportError(err)
 			return err
 		}
 
-		if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, name, time.Duration(5*time.Minute), "component"); err != nil {
-			kuskui.PrintError("failed installing Envoyfleets")
+		// if err := utils.WaitKuskCRDsReady(cmd.Context()); err != nil {
+		// 	kuskui.PrintError("❌ failed installing Kusk")
+		// 	reportError(err)
+		// 	return err
+		// }
+
+		if err := utils.WaitAPIServiceReady(cmd.Context(), c); err != nil {
+			kuskui.PrintError("❌ failed installing Kusk")
 			reportError(err)
 			return err
 		}
 
 		if !noEnvoyFleet {
 			kuskui.PrintStart("installing Envoyfleets...")
-			if err := applyf(filepath.Join(dir, manifests_dir, "fleets.yaml")); err != nil {
+			manifest, err := os.ReadFile(filepath.Join(dir, manifests_dir, "fleets.yaml"))
+			if err != nil {
 				kuskui.PrintError("failed installing Envoyfleets")
 				reportError(err)
 				return err
 			}
 
-			if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "envoy", time.Duration(5*time.Minute), "component"); err != nil {
+			fleet := &kuskv1.EnvoyFleet{}
+			if err := yaml.Unmarshal(manifest, fleet); err != nil {
 				kuskui.PrintError("failed installing Envoyfleets")
 				reportError(err)
 				return err
 			}
 
+			retry.Do(
+				func() error {
+					return c.Create(cmd.Context(), fleet, &client.CreateOptions{})
+				},
+				retry.DelayType(func(n uint, err error, config *retry.Config) time.Duration {
+
+					return time.Duration(10 * time.Second)
+				}))
+
+			if err != nil {
+				kuskui.PrintError("failed installing Envoyfleets")
+				reportError(err)
+				return err
+			}
+
+			if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "envoy", time.Duration(5*time.Minute), "component"); err != nil {
+				kuskui.PrintError("failed installing Envoyfleets")
+				reportError(err)
+				return err
+			}
 		} else {
 			return nil
 		}
 
 		if !noApi {
 			kuskui.PrintStart("installing API Server...")
-			if err := applyf(filepath.Join(dir, manifests_dir, "apis.yaml")); err != nil {
+			manifest, err := os.ReadFile(filepath.Join(dir, manifests_dir, "api_server_api.yaml"))
```

Comment:
I don't understand why you are applying this in the `!noApi` block.

Shouldn't it be the reverse?